File: app/src/main/java/com/ytempest/wanandroid/activity/main/MainFragConstructor.java
package com.ytempest.wanandroid.activity.main;

import androidx.fragment.app.Fragment;

import com.ytempest.tool.helper.ExpandFragHelper;
import com.ytempest.wanandroid.activity.main.home.HomeFrag;
import com.ytempest.wanandroid.activity.main.knowledge.KnowledgeFrag;
import com.ytempest.wanandroid.activity.main.navigation.NavigationFrag;
import com.ytempest.wanandroid.activity.main.project.ProjectFrag;

/**
 * @author heqidu
 * @since 2020/6/24
 */
public class MainFragConstructor implements ExpandFragHelper.FragConstructor {
    @Override
    public Fragment createFrag(@MainTab.Id int fragId) {
        switch (fragId) {
            case MainTab.Id.HOME:
                return new HomeFrag();

            case MainTab.Id.KNOWLEDGE:
                return new KnowledgeFrag();

            case MainTab.Id.NAVIGATION:
                return new NavigationFrag();

            case MainTab.Id.PROJECT:
                return new ProjectFrag();

            default:
                throw new IllegalStateException("Not found the fragment for id : " + fragId);
        }
    }
}

File: app/src/main/java/com/ytempest/wanandroid/activity/main/MainFragConstructor.kt
package com.ytempest.wanandroid.activity.main

import androidx.fragment.app.Fragment
import com.ytempest.wanandroid.activity.main.home.HomeFrag
import com.ytempest.wanandroid.activity.main.knowledge.KnowledgeFrag
import com.ytempest.wanandroid.activity.main.navigation.NavigationFrag
import com.ytempest.wanandroid.activity.main.project.ProjectFrag
import com.ytempest.tool.helper.ExpandFragHelper

/**
 * @author heqidu
 * @since 21-2-8
 */
class MainFragConstructor : ExpandFragHelper.FragConstructor {

    override fun createFrag(@MainTab.Id fragId: Int): Fragment {
        return when (fragId) {
            MainTab.Id.HOME -> HomeFrag()
            MainTab.Id.KNOWLEDGE -> KnowledgeFrag()
            MainTab.Id.NAVIGATION -> NavigationFrag();
            MainTab.Id.PROJECT -> ProjectFrag();
            else -> throw IllegalStateException("Not found the fragment for id : $fragId")
        }
    }
}